### PR TITLE
Add solution & readme to solution/pointer_and_structs

### PR DESCRIPTION
**Pull request by busin11**
1- Add file: pointer_and_structs.go
2- Edit README.md

### CODE REVIEW

File: pointers_and_structs/README.md [hunk 1]
```diff
@@ -30,3 +30,69 @@ Original Rectangle: Width = 5, Height = 3
...
+
+
+
+## How it runs
+
+
+
+Execute go code:
+
+```bash
```
Comment (tul1) [inline pointers_and_structs/README.md:60]: why have you change the format of the readme here?

File: pointers_and_structs/pointer_structs.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+
+type Rectangle struct {
+	width, height float64
+}
+
+
+
+func ScaleRectByPointer(r1 *Rectangle) {
```
Comment (tul1) [inline pointers_and_structs/pointer_structs.go:15]: can you turn `ScaleRectByPointer` and `ScaleRectByValue`into methods of the class?

File: pointers_and_structs/README.md (hunk 1 above)
Comment (busin11) [reply pointers_and_structs/README.md:60]: i'am not set yet to use git correctly, will train on my side again

File: pointers_and_structs/pointer_structs.go (hunk 2 above)
Comment (busin11) [reply pointers_and_structs/pointer_structs.go:15]: func (r1 *Rectangle) ScaleRectByPointer() {
func (r1 Rectangle) ScaleRectByValue() {

func main() {
        r1 := Rectangle{3, 3}
        fmt.Printf("\n Original Rectangle: Width = %.1f , Height = %.1f", r1.width, r1.height)

        r1.ScaleRectByValue()
        fmt.Printf("\n Scaled by Value (No Change): Width = %.1f , Height = %.1f", r1.width, r1.height)

        r1.ScaleRectByPointer()
        fmt.Printf("\n Scaled by Pointer (Changed): Width = %.1f , Height = %.1f", r1.width, r1.height)
}

**Comment by busin11** [comment]
I have to ask you a question in order to use git correctly, might we can discuss next week ?

File: pointers_and_structs/README.md [hunk 3]
```diff
@@ -30,3 +30,69 @@ Original Rectangle: Width = 5, Height = 3
 Scaled by Value (No Change): Width = 5, Height = 3
 Scaled by Pointer (Changed): Width = 10, Height = 6
 ```
+
+## How it works
+
+
+
+1- Define Rectangle struct with two float64 var : width, height
```
Comment (tul1) [inline pointers_and_structs/README.md:38]:
```suggestion
1- Define Rectangle struct with two float64 var are: `width`, `height`
```

File: pointers_and_structs/README.md [hunk 4]
```diff
@@ -30,3 +30,69 @@ Original Rectangle: Width = 5, Height = 3
 Scaled by Value (No Change): Width = 5, Height = 3
 Scaled by Pointer (Changed): Width = 10, Height = 6
 ```
+
+## How it works
+
+
+
```
Comment (tul1) [inline pointers_and_structs/README.md:37]:
```suggestion
```

File: pointers_and_structs/README.md [hunk 5]
```diff
@@ -30,3 +30,69 @@ Original Rectangle: Width = 5, Height = 3
 Scaled by Value (No Change): Width = 5, Height = 3
 Scaled by Pointer (Changed): Width = 10, Height = 6
 ```
+
+## How it works
+
+
```
Comment (tul1) [inline pointers_and_structs/README.md:36]:
```suggestion
```

File: pointers_and_structs/README.md [hunk 6]
```diff
@@ -30,3 +30,69 @@ Original Rectangle: Width = 5, Height = 3
 Scaled by Value (No Change): Width = 5, Height = 3
 Scaled by Pointer (Changed): Width = 10, Height = 6
 ```
+
+## How it works
+
```
Comment (tul1) [inline pointers_and_structs/README.md:35]:
```suggestion
```

**Comment by tul1** [comment]
> I have to ask you a question in order to use git correctly, might we can discuss next week ?

definetely

**Comment by busin11** [comment]
Will ask you the question tomorrow, i cleaned commit history. Normally BR is mergeable.

File: pointers_and_structs/pointer_and_structs.go [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+type Rectangle struct {
+        width, height float64
+}
+
+func (r1 *Rectangle) ScaleRectByPointer() {
+        r1.height = 1.6
+        r1.width = 1.6
+}
+
+func (r1 Rectangle) ScaleRectByValue() {
```
Comment (tul1) [inline pointers_and_structs/pointer_and_structs.go:14]: can you turns this method into func

**Review by tul1 — APPROVED** [review]

**Pull request merged**